src/main/java/com/ita/if103java/ims/service/impl/ItemServiceImpl.java

```diff
@@ -67,75 +75,113 @@ public ItemServiceImpl(ItemDtoMapper itemDtoMapper, SavedItemDtoMapper savedItem
+            }
             return savedItemDto;
+        } else {
+            eventService.create(new Event("Not enough capacity! Capacity " + warehouseDao.findById(itemTransaction.getDestinationWarehouseId()).getCapacity() +
+                " in Warehouse " + warehouseDao.findById(itemTransaction.getDestinationWarehouseId()).getName(),
+                user.getUser().getAccountId(),
+                itemTransaction.getDestinationWarehouseId(), user.getUser().getId(), EventName.LOW_SPACE_IN_WAREHOUSE
+                , null));
+            throw new ItemNotEnoughCapacityInWarehouseException("Can't add savedItemDto in warehouse because it " +
+                "doesn't  " +
+                "have enough capacity {warehouse_id = " + itemTransaction.getDestinationWarehouseId() + "}");
         }
-        throw new SavedItemNotFoundException("Failed to get savedItem during `create` {account_id = " + itemTransaction.getItemDto().getAccountId() + "}");
+
+
     }
 
-    private boolean isEnoughCapacityInWarehouse(ItemTransactionRequestDto itemTransaction, Function<ItemTransactionRequestDto, Long> function) {
-        float volume = getVolumeOfPassSavedItems(itemTransaction) + itemTransaction.getQuantity() * itemTransaction.getItemDto().getVolume();
+
+    private boolean isEnoughCapacityInWarehouse(ItemTransactionRequestDto itemTransaction, Long accountId) {
+        float volume =
+            getVolumeOfPassSavedItems(itemTransaction, accountId) + itemTransaction.getQuantity() * itemTransaction.getItemDto().getVolume();
         return warehouseDao.findById(itemTransaction.getDestinationWarehouseId()).getCapacity() >= volume;
     }
 
-    private float getVolumeOfPassSavedItems(ItemTransactionRequestDto itemTransaction) {
+    private float getVolumeOfPassSavedItems(ItemTransactionRequestDto itemTransaction, Long accountId) {
         float volumePassSavedItems = 0;
         Long warehouseId = itemTransaction.getDestinationWarehouseId();
 
         if (savedItemDao.existSavedItemByWarehouseId(warehouseId)) {
             for (SavedItem savedItem : savedItemDao.findSavedItemByWarehouseId(warehouseId)) {
-                Item item = itemDao.findItemById(savedItem.getItemId());
+                Item item = itemDao.findItemById(savedItem.getItemId(), accountId);
                 volumePassSavedItems += savedItem.getQuantity() * item.getVolume();
-                System.out.println(volumePassSavedItems + item.getName() + item.getVolume() + savedItem.getQuantity());
             }
         }
-        System.out.println(volumePassSavedItems);
         return volumePassSavedItems;
     }
 
 
-    private boolean isLowSpaceInWarehouse(ItemTransactionRequestDto itemTransaction) {
-        if (getVolumeOfPassSavedItems(itemTransaction) == 0) {
+    private boolean isLowSpaceInWarehouse(ItemTransactionRequestDto itemTransaction, Long accountId) {
+        float volume = getVolumeOfPassSavedItems(itemTransaction, accountId);
+        if (volume == 0) {
```

Comment:
here we need to calculate load of the warehouse and not of volume of PassSavedItems, because we already saved them

Reply:
My bad, It's already implemented. close this comment